realm/realm-library/src/androidTest/kotlin/io/realm/ParameterizedDictionaryTests.kt

```diff
@@ -0,0 +1,371 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm
+
+import androidx.test.platform.app.InstrumentationRegistry
+import io.realm.entities.DogPrimaryKey
+import io.realm.rule.TestRealmConfigurationFactory
+import org.bson.types.Decimal128
+import org.bson.types.ObjectId
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import java.util.*
+
+/**
+ * Dictionary tests. It uses [Parameterized] tests for all possible combinations of
+ * [RealmDictionary] types (i.e. all primitive Realm types (see [DictionarySupportedType]) plus
+ * [RealmModel] and [Mixed] (and in turn all possible types supported by Mixed) in both `managed`
+ * and `unmanaged` modes.
+ *
+ * In order to streamline the testing for managed dictionaries we use Kotlin's reflection API
+ * `KFunction1` and `KFunction2`. These two methods provide access to the Java getters and setters
+ * used to work with each dictionary field.
+ */
+@RunWith(Parameterized::class)
+class ParameterizedDictionaryTests(
```

Comment:
When you have this pattern of almost delegating directly through to a specific incoming parameter you can use inheritance by delegation to automatically. Don't know now to put annotations on the individual methods without explicitly defining them though 😞